src/PFire.Core/Protocol/MessageSerializer.cs

```diff
@@ -41,8 +41,26 @@ public static IMessage Deserialize(BinaryReader reader, IMessage messageBase)
 
                 var value = XFireAttributeFactory.Instance.GetAttribute(attributeType).ReadValue(reader);
 
-                var field = fieldInfo.Where(a => a.GetCustomAttribute<XMessageField>() != null)
-                                     .FirstOrDefault(a => a.GetCustomAttribute<XMessageField>()?.Name == attributeName);
+                var field = fieldInfo
+                    .Where(a => a.GetCustomAttribute<XMessageField>() != null)
+                    .FirstOrDefault(a =>
+                    {
+                        var attribute = a.GetCustomAttribute<XMessageField>();
+                        if (attribute != null)
+                        {
+                            if (attributeName.Length == 1)
```

Comment:
So, the attribute name gets put into utf8 encoding.
Which is fine for strings, but for single byte attributes, the encoding part breaks certain hex values and causes it to return unexpected and unwanted results for Inbound packets with single bytes of certain values. 
Chatroom packets suffered the worst with this, so I wrote this to return the exact byte given back so that it all smoothly gets decoded fine if the value is a single byte. It's ugly but it made it work.

Reply:
Oh yeah, that makes a lot of sense. Instead of this special logic, I wonder if we can just use a 1 byte string encoding like ASCII instead of UTF8. We don't need any fancy characters for our field attribute names.

Reply:
As long as ASCII doesn't try to encode say 0x01 and treat it as a special byte like utf8 did, that's my only fear.

Reply:
I reverted this change to how it was and am able to update my status to trigger `StatusChange` (which has the byte type attribute) and it works fine.

Do you have a specific example where you ran into your issue?

Reply:
My memory isn't the best but I overlooked some notes I took on this. 

Most bytes should work, especially if they are near the letters portion of a utf8 charmap. 

One of the examples I can think of is when the problem arised when inbound packets for User Group Member adding and User Request Advanced Info, which both contain 0x01 as an attribute name would deserialize and produce an attribute of \u0001 instead of the expected 0x01. 

The problem arises when it hits specific utf8 special characters, this would fix the troublesome bytes that would break during incoming packet deserialization.

Reply:
If we put back the MessageSerializer.Deserialize to this

```
var attributeName = GetAttributeName(reader, messageType);
var field = fieldInfo
    .Where(a => a.GetCustomAttribute<XMessageField>() != null)
    .FirstOrDefault(a => a.GetCustomAttribute<XMessageField>()?.Name == attributeName);
```

Put back the MessageSerializer.GetAttributeName to return the encoded string
```
var count = messageTypeSet.Contains(messageType) ? 1 : reader.ReadByte();
var readBytes = reader.ReadBytes(count);
return Encoding.UTF8.GetString(readBytes);
```

Uncomment the call to the other constructor in `XMessageField` here to set the name property
```
public XMessageField(params byte[] name) 
    : this(Encoding.UTF8.GetString(name))
{
    NonTextualName = true;
    NameAsBytes = name;
}
```

Then the  `.FirstOrDefault` above would act like this no?
`a.GetCustomAttribute<XMessageField>()?.Name` would be "\u0001"
`attributeName` would be "\u0001" as well

So it should match fine.